### ISSUE TITLE
Add `has:expiration` to search terms

### ISSUE THREAD

**Issue by ledgeri** [keepassxreboot/keepassxc#10584]
Hi!

I wanted to list entries matching the title-mentioned criteria, but the best what i found is the `!is:expired` search-term, but it is not perfect, because it also list everithing what is not expired, so as soon as i sort the results by an other parameter (name or group-name, etv) it mixes them all.

Is there a more direct way?

**Comment by droidmonkey** [comment]
Yes, show the "Expires" column (right click the entry list header row). Then sort by that column. Entries with "Never" don't have the checkbox checked.

**Comment by ledgeri** [comment]
> Yes, show the "Expires" column (right click the entry list header row). Then sort by that column. Entries with "Never" don't have the checkbox checked.

That solution does not satisfy the "show **only** `expiration not set`" cliteria, and by that the "sort only those by other column" wish/need.

But ok,  if i am getting more annoyed by this i will make a feature request in the future.

**Comment by droidmonkey** [comment]
You'd want something like `has:expiration` as a search term pair to return the desired result. I have been consider a couple of "has:" pairs, like `has:totp` for example.

**Comment by ledgeri** [comment]
> You'd want something like `has:expiration` as a search term pair to return the desired result. I have been consider a couple of "has:" pairs, like `has:totp` for example.

Yes, similar things! A "search-term deatail pass" would be nice, to add extras. But low priority, for sure. Now enough to me knowing you had similar ideas, so it can attached to one, when the time is right for implementing these!
Cheers!

**Comment by ledgeri** [comment]
I sat on it, and i could suggest an more complex version (just as an idea): #11117
